gcc/rust/ast/rust-ast-full-test.cc

```diff
@@ -379,22 +379,20 @@ VisItem::as_string () const
 std::string
 Module::as_string () const
 {
-  std::string vis_item = VisItem::as_string ();
+  std::string str = VisItem::as_string () + "mod " + module_name;
 
-  return vis_item + "mod " + module_name;
-}
-
-std::string
-ModuleBodied::as_string () const
-{
-  // get module string for "[vis] mod [name]"
-  std::string str = Module::as_string ();
+  // Return early if we're dealing with an unloaded module as their body resides
+  // in a different file
+  if (kind == ModuleKind::UNLOADED)
+    return str + "\n no body (reference to external file)\n";
 
   // inner attributes
   str += append_attributes (inner_attrs, INNER);
 
   // items
   str += "\n items: ";
+
+  // This can still happen if the module is loaded but empty, i.e. `mod foo {}`
   if (items.empty ())
```

Comment:
It might be worth emitting a "not loaded" instead of empty for that case

Reply:
Yeah I was wondering about that, but since the distinction is made at parsing based on whether the next token is a curly brace or a semicolon, I don't know if it's possible. Unless I'm misunderstanding your comment?